### DISCUSSION TITLE
Dockerize Synthea

### DISCUSSION THREAD

**Discussion by npagare** [synthetichealth/synthea#1432] [Q&A]
Hello All,

Has anyone created docker container for Synthea?

Thank you,
Nilesh

**Comment by jawalonoski** [comment]
See https://github.com/synthetichealth/synthea/pull/1406

That is at least one example.

**Comment by npagare** [comment]
Thank you @jawalonoski 

Any thoughts on when will it be merged in the main branch?

**Reply by jawalonoski** [reply]
It doesn't look like it is going to be merged, since the reviewer requested changes, and the submitter doesn't seem like they are going to address them.

**Comment by dehall** [comment]
See also the Synthea Customizer at https://synthetichealth.github.io/spt/#/customizer , it provides some options to customize synthea and at the end you can download a dockerfile with your own specific customizations

**Reply by npagare** [reply]
Hi @dehall, thank you for your suggestion.

My goal is to put a docker build process against the main branch (every time it updates) and then run Synthea based on externalized parameters provided via environmental variables to create output as mentioned above.

The goal is to create data for a hospital organization operating at multiple locations (City, State, country).
So, customization around that by providing organization data will be one use case to work with.

Thank you,
Nilesh

**Reply by dehall** [reply]
Hm ok, I guess I'm not sure I understand how (or why) to pin it against updates to the main branch, but it seems like your best bet would be to create your own Dockerfile. Since Synthea has so many command line options, I suspect any generic docker image is either going to be severely limited or overly complex. The "base" dockerfile from the customizer fetches the prebuilt JAR from github and runs it, so you could start with that and add the specific customizations you need:

```
FROM eclipse-temurin:20-jdk

# add --insecure for a quick fix if your network requires certificates
# -L means follow redirects
RUN curl -L -O https://github.com/synthetichealth/synthea/releases/download/master-branch-latest/synthea-with-dependencies.jar
 
CMD java -jar synthea-with-dependencies.jar
```
